circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/DropTableService.java

```diff
@@ -56,13 +100,52 @@ public void removeTableParamsAndDrop(
       }
       client.alter_table(databaseName, tableName, table);
     }
-    LOG
-        .info("Dropping table '{}.{}'.", table.getDbName(), table.getTableName());
-    client.dropTable(table.getDbName(), table.getTableName(), false, true);
+    LOG.info("Dropping table '{}.{}'.", databaseName, tableName);
+    client.dropTable(databaseName, tableName, false, true);
+  }
+
+  private void deleteData(DataManipulationClient dataManipulationClient, String replicaDataLocation) {
+    try {
+      LOG.info("Dropping table data from location: {}.", replicaDataLocation);
+      boolean dataDeleted = dataManipulationClient.delete(replicaDataLocation);
+      LOG.info("Data deleted: {}.", dataDeleted);
+    } catch (IOException e) {
+      LOG.info("Could not drop replica table data at location:{}.", replicaDataLocation);
+    }
+  }
+
+  private void deletePartitionData(
+      DataManipulationClient dataManipulationClient,
+      String replicaTableLocation,
+      List<String> replicaPartitionLocations) {
+    try {
+      LOG.info("Dropping partition data from base location: {}.", replicaTableLocation);
+      for (String location : replicaPartitionLocations) {
+        boolean deleted = dataManipulationClient.delete(location);
+        LOG.debug("Attempted to delete data from location: {}. Successful deletion = {}.", location, deleted);
+      }
+    } catch (IOException e) {
+      LOG.info("Could not drop replica partition data at location:{}.", replicaTableLocation);
+    } catch (UnsupportedOperationException e) {
+      LOG.info(e.getMessage());
+    }
   }
 
   private boolean isExternal(Map<String, String> tableParameters) {
     CaseInsensitiveMap caseInsensitiveParams = new CaseInsensitiveMap(tableParameters);
     return IS_EXTERNAL.equalsIgnoreCase((String) caseInsensitiveParams.get(EXTERNAL_KEY));
   }
+
+  private List<String> getPartitionLocations(CloseableMetaStoreClient client, Table table) {
+    List<String> locations = new ArrayList<>();
+    try {
+      PartitionIterator partitionIterator = new PartitionIterator(client, table, BATCH_SIZE);
+      while (partitionIterator.hasNext()) {
+        locations.add(partitionIterator.next().getSd().getLocation());
+      }
+    } catch (TException e) {
+      LOG.info("Could not list partitions for {}.{}.", table.getDbName(), table.getTableName());
```

Comment:
Log the exception.

Also, should we swallow all these exceptions with warnings or should we throw them and fail the replication?

Reply:
I don't mind. It technically wont affect the replication, but you could end up with a lot of orphaned data which wont be cleaned up.

Reply:
I think all exceptions in this class except ones we can recover from should be thrown and stop the replication. Logging them serves little purpose, no one is going to look through the log if the end replication reports OK. If the purpose is alerting the user to manually clean up, best to just fail. If this becomes unfeasible e.g. lots of failures we can build retry mechanics etc.. but out of scope now.